File: Java x EDAI/Colas/Cola.java
public class Cola {

	public Nodo inicio, fin;
	public int tamaño;

	// Constructor de la Cola
	public Cola(){
		this.inicio = null;
		this.fin = null;
		this.tamaño = 0;
	}

	public static void main(String[] args){
		Cola cola = new Cola();
		Nodo tmp;

		Nodo a = new Nodo(3);
		Nodo b = new Nodo(7);
		Nodo c = new Nodo(1);
		Nodo d = new Nodo(5);
		Nodo e = new Nodo(7);
		Nodo f = new Nodo(8);
		Nodo g = new Nodo(4);

		System.out.println("Agregando elementos");
		cola.push(a);
		tmp = cola.peek(); System.out.println(tmp.valor);
		cola.push(b);
		tmp = cola.peek(); System.out.println(tmp.valor);
		cola.push(c);
		cola.push(d);
		cola.push(e);
		tmp = cola.peek(); System.out.println(tmp.valor);
		cola.push(f);
		cola.push(g);
		tmp = cola.peek(); System.out.println(tmp.valor);

		cola.imprimirCola();

		System.out.println("Removiendo elementos");
		tmp = cola.pop();
		tmp = cola.pop(); System.out.println(tmp.valor);
		tmp = cola.pop();
		tmp = cola.pop(); System.out.println(tmp.valor);

		cola.imprimirCola();

		tmp = cola.pop();
		tmp = cola.pop(); System.out.println(tmp.valor);
		tmp = cola.pop();

		cola.imprimirCola();

		tmp = cola.pop();
	}

	// Agregar un elemento a la Cola
	public void push(Nodo nodo){
		// Se inserta el primer elemento
		if (inicio==null && fin==null && tamaño==0){
			inicio = nodo;
			fin = nodo;
		} else { // La Cola ya contiene elementos
			fin.siguiente = nodo;
			fin = nodo;
		}

		tamaño++;
	}

	// Remueve el elemento que se encuentra al inicio de la Cola
	public Nodo pop(){
		// Si no hay elementos en la Cola, no se puede extraer
		if (tamaño==0 && inicio==null && fin==null){
			System.out.println("\nError: no hay elementos en la Cola\n");
		} else { // Si hay un elemento al inicio, se extrae
			Nodo tmp = inicio;
			inicio = inicio.siguiente;

			// Si se remueve el último elemento, la cola queda vacía
			if (tamaño==1){
				fin = null;
			}

			tamaño--;

			return tmp;
		}

		return null;
	}

	// Muestra el elemento al inicio de la Cola sin removerlo
	public Nodo peek(){
		// Si no hay elementos en la Cola, no se puede visualizar
		if (tamaño==0 && inicio==null && fin==null){
			System.out.println("\nError: no hay elementos en la Cola\n");
		} else { // Si hay un elemento al inicio, se visualiza
			Nodo tmp = inicio;

			return tmp;
		}

		return null;
	}

	// Muestra el contenido de la Cola
	public void imprimirCola(){
		// Si no hay elementos en la Cola, no se imprime
		if (tamaño==0 && inicio==null && fin==null){
			System.out.println("\nCola sin elementos");
		} else { // Si hay elementos en la Cola, se muestran en pantalla
			System.out.println("\nCola");
			System.out.println("Inicio: " + this.inicio.valor);
			System.out.println("Fin: " + this.fin.valor);
			System.out.println("Tamaño: " + this.tamaño);
			System.out.print("Contenido: ");

			Nodo tmp = inicio;

			while (tmp != null){
				System.out.print(tmp.valor + ", ");
				tmp = tmp.siguiente;
			}
			System.out.println("\n");
		}
	}

}

File: EDAI x Python/Colas/Cola.py
from Nodo import Nodo

class Cola:
	
	# Constructor de la Cola
    def __init__(self):
        self.inicio = None
        self.fin = None
        self.tamaño = 0

    # Agregar un elemento a la Cola
    def push(self, nodo):
    	# Se inserta el primer elemento
        if self.inicio is None and self.fin is None and self.tamaño == 0:
            self.inicio = nodo
            self.fin = nodo
        # La Cola ya contiene elementos
        else:
            self.fin.siguiente = nodo
            self.fin = nodo
        self.tamaño += 1

    # Remueve el elemento que se encuentra al inicio de la Cola
    def pop(self):
    	# Si no hay elementos en la Cola, no se puede extraer
        if self.tamaño == 0 and self.inicio is None and self.fin is None:
            print("\nError: no hay elementos en la Cola\n")
        # Si hay un elemento al inicio, se extrae
        else:
            tmp = self.inicio
            self.inicio = self.inicio.siguiente

            # Si se remueve el último elemento, la cola queda vacía
            if self.tamaño == 1:
                self.fin = None
            self.tamaño -= 1
            return tmp
        return None

    # Muestra el elemento al inicio de la Cola sin removerlo
    def peek(self):
    	# Si no hay elementos en la Cola, no se puede visualizar
        if self.tamaño == 0 and self.inicio is None and self.fin is None:
            print("\nError: no hay elementos en la Cola\n")
        # Si hay un elemento al inicio, se visualiza
        else:
            return self.inicio
        return None

    # Muestra el contenido de la Cola
    def imprimirCola(self):
    	# Si no hay elementos en la Cola, no se imprime
        if self.tamaño == 0 and self.inicio is None and self.fin is None:
            print("\nCola sin elementos")
        # Si hay elementos en la Cola, se muestran en pantalla
        else:
            print("\nCola")
            print("Inicio:", self.inicio.valor)
            print("Fin:", self.fin.valor)
            print("Tamaño:", self.tamaño)
            print("Contenido:", end=" ")
            tmp = self.inicio
            while tmp is not None:
                print(tmp.valor, end=", ")
                tmp = tmp.siguiente
            print("\n")

    def main(self):
	    cola = Cola()
	    tmp = None

	    a = Nodo(3)
	    b = Nodo(7)
	    c = Nodo(1)
	    d = Nodo(5)
	    e = Nodo(7)
	    f = Nodo(8)
	    g = Nodo(4)

	    print("Agregando elementos")
	    cola.push(a)
	    tmp = cola.peek()
	    print(tmp.valor)
	    cola.push(b)
	    tmp = cola.peek()
	    print(tmp.valor)
	    cola.push(c)
	    cola.push(d)
	    cola.push(e)
	    tmp = cola.peek()
	    print(tmp.valor)
	    cola.push(f)
	    cola.push(g)
	    tmp = cola.peek()
	    print(tmp.valor)

	    cola.imprimirCola()

	    print("Removiendo elementos")
	    tmp = cola.pop()
	    tmp = cola.pop()
	    print(tmp.valor)
	    tmp = cola.pop()
	    tmp = cola.pop()
	    print(tmp.valor)

	    cola.imprimirCola()

	    tmp = cola.pop()
	    tmp = cola.pop()
	    print(tmp.valor)
	    tmp = cola.pop()

	    cola.imprimirCola()

	    tmp = cola.pop()


# Ejecutando el objeto
programa = Cola()
programa.main()